src/shapes/IText/ITextBehavior.ts

```diff
@@ -4,7 +4,7 @@ import type {
   TPointerEventInfo,
 } from '../../EventTypeDefs';
 import { Point } from '../../Point';
-import type { FabricObject } from '../Object/FabricObject';
```

Comment:
Gets later used in `onDeselect`. This means the problem occurred already in fabric itself, but the fix was not to use InteractiveObject here as well. The fix is to remove the type specialization in InteractiveObject's `onDeselect`

Reply:
This is incorrect, what is the problem it is causing?

Reply:
<img width="691" alt="Screenshot 2023-09-10 at 11 26 22" src="https://github.com/fabricjs/fabric.js/assets/10067273/997792e9-d6e6-4c6d-9dbe-fa9c1284fa9c">

As I said, it's used for onDeselect